sound/soc/sof/intel/hda.c

```diff
@@ -420,6 +420,7 @@ static bool hda_sdw_check_wakeen_irq(struct snd_sof_dev *sdev)
 	if (!(interface_mask & BIT(SOF_DAI_INTEL_ALH)))
 		return false;
 
+	chip = get_chip_info(sdev->pdata);
```

Comment:
:astonished:

Reply:
I'm puzzled that this worked until now :)

Reply:
yeah, me too

Reply:
I mean: gcc, sparse should have spotted this with ease, but it is not. Not to mention the reviewers!

Reply:
yeah two big fails for the prize of one
a) the jack detection was completely broken and we didn't notice and
b) we didn't even compile with clang.

I don't know how we are going to send people on Mars ...